tests/manage/pv_services/test_resource_deletion_during_pod_pvc_deletion.py

```diff
@@ -0,0 +1,323 @@
+        initial_num_of_pods = len(get_all_pods(namespace=self.namespace))
+        initial_num_of_pvc = len(
+            get_all_pvcs(namespace=self.namespace)['items']
+        )
+
+        # Fetch PV names
+        pv_objs = []
+        for pvc_obj in self.pvc_objs:
+            pvc_obj.reload()
+            pv_objs.append(pvc_obj.backed_pv_obj)
+
+        # Fetch volume details from pods for the purpose of verification
+        node_pv_dict = {}
+        for pod_obj in self.pod_objs:
+            pod_info = pod_obj.get()
+            node = pod_info['spec']['nodeName']
+            pvc = pod_info['spec']['volumes'][0]['persistentVolumeClaim']['claimName']
+            for pvc_obj in self.pvc_objs:
+                if pvc_obj.name == pvc:
+                    pvc_obj.reload()
+                    pv = pvc_obj.backed_pv
+                    break
+            if node in node_pv_dict:
+                node_pv_dict[node].append(pv)
+            else:
+                node_pv_dict[node] = [pv]
+
+        # Do setup for running IO on pods
+        log.info("Setting up pods for running IO")
+        for pod_obj in self.pod_objs:
+            pod_obj.workload_setup(storage_type='fs')
+        log.info("Setup for running IO is completed on pods")
+
+        # Start IO on each pod
+        log.info("Starting IO on pods")
+        for pod_obj in self.pod_objs:
+            pod_obj.run_io(storage_type='fs', size=f'{self.pvc_size - 1}G')
+        log.info("IO started on all pods.")
+
+        # Start deleting pods
+        pod_bulk_delete = executor.submit(self.delete_pods)
+
+        if operation_to_disrupt == 'delete_pods':
+            ret = self.verify_resource_deletion(
+                get_all_pods, initial_num_of_pods
+            )
+            assert ret, "Wait timeout: Pods are not being deleted."
+            logging.info(
+                f"Pods deletion has started."
+            )
+            disruption.delete_resource()
+
+        pods_deleted = pod_bulk_delete.result()
+
+        assert pods_deleted, "Deletion of pods failed."
+
+        # Verify pods are deleted
+        for pod_obj in self.pod_objs:
+            assert pod_obj.ocp.wait_for_delete(pod_obj.name), (
+                f"Pod {pod_obj.name} is not deleted"
+            )
+        logging.info("Verified: Pods are deleted.")
+
+        # Verify that the mount point is removed from nodes after deleting pod
+        for node, pvs in node_pv_dict.items():
+            cmd = f'oc debug nodes/{node} -- df'
+            df_on_node = run_cmd(cmd)
+            for pv in pvs:
+                assert pv not in df_on_node, (
+                    f"{pv} is still present on node {node} after "
+                    f"deleting the pods."
+                )
+
+        # Fetch image uuid associated with PVCs
+        pvc_uuid_map = {}
+        for pvc_obj in self.pvc_objs:
+            pvc_uuid_map[pvc_obj.name] = pvc_obj.image_uuid
+
+        # Start deleting PVCs
```

Comment:
add more loggers please.
It will be very helpful to debug with more logs

Reply:
Added more logs